Sources/SmithyReadWrite/DocumentWriter.swift

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.Data
+
```

Comment:
This type was moved in from `ClientRuntime`, and was made generic to any `SmithyWriter`.